test/common/routes_exist.go

```diff
@@ -0,0 +1,224 @@
+
+import (
+	"fmt"
+	"testing"
+
+	goctx "context"
+
+	routev1 "github.com/openshift/api/route/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type expectedRoute struct {
+	Name            string
+	isTLS           bool
+	ServiceName     string
+	IsGeneratedName bool
+}
+
+var expectedRoutes = map[string][]expectedRoute{
+	"3scale": []expectedRoute{
+		expectedRoute{
+			Name:  "backend",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:            "zync-3scale-api-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "apicast-staging",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-api-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "apicast-production",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-master-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-master",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-provider-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-developer",
+		},
+		expectedRoute{
+			Name:            "zync-3scale-provider-",
+			IsGeneratedName: true,
+			isTLS:           true,
+			ServiceName:     "system-provider",
+		},
+	},
+
+	"amq-online": []expectedRoute{
+		expectedRoute{
+			Name:  "console",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "standard-authservice",
+			isTLS: true,
+		},
+	},
+
+	"codeready-workspaces": []expectedRoute{
+		expectedRoute{
+			Name:  "codeready",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "devfile-registry",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "plugin-registry",
+			isTLS: true,
+		},
+	},
+
+	"middleware-monitoring-operator": []expectedRoute{
+		expectedRoute{
+			Name:  "alertmanager-route",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "grafana-route",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "prometheus-route",
+			isTLS: true,
+		},
+	},
+
+	"rhsso": []expectedRoute{
+		expectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		expectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+
+	"solution-explorer": []expectedRoute{
+		expectedRoute{
+			Name:  "tutorial-web-app",
+			isTLS: true,
+		},
+	},
+
+	"ups": []expectedRoute{
+		expectedRoute{
+			Name:  "ups-unifiedpush-proxy",
+			isTLS: true,
+		},
+	},
+
+	"user-sso": []expectedRoute{
+		expectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		expectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+}
+
+const (
+	intlyNamespacePrefix = "redhat-rhmi-"
+)
+
+// TestIntegreatlyRoutesExist tests that the routes for all the products are created
+func TestIntegreatlyRoutesExist(t *testing.T, ctx *TestingContext) {
+	var err error
+
+	for product, routes := range expectedRoutes {
+		for _, expectedRoute := range routes {
+			err = checkRoute(t, ctx, product, expectedRoute)
+
+			if err != nil {
+				t.Fatalf("Failed checking route %v for product %v: %v",
```

Comment:
I will suggest using `t.Errorf`, the error will be logged, the test marked as failed but the execution of the test will proceed and we could catch other problems 
```suggestion
				t.Errorf("Failed checking route %v for product %v: %v",
```

Reply:
Makes sense, I'll change that in a moment. Thanks @b1zzu !